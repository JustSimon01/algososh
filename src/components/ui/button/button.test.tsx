import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './button';

describe('Тесты кнопки', () => {
  test('Кнопка без текста', ()=>{
    render(<Button/>);
    const btn = screen.getByRole('button');
    expect(btn).toMatchSnapshot();
  })

  test('Кнопка c текстом', ()=>{
    render(<Button text='текст кнопки'/>);
    const btn = screen.getByRole('button');
    expect(btn).toMatchSnapshot();
  })

  test('Кнопка заблокированна', ()=>{
    render(<Button disabled={true}/>);
    const btn = screen.getByRole('button');
    expect(btn).toMatchSnapshot();
  })

  test('Кнопка c индикацией загрузки', ()=>{
    render(<Button isLoader={true}/>);
    const btn = screen.getByRole('button');
    expect(btn).toMatchSnapshot();
  })

  test('Проверка вызова колбека', ()=>{
    const someFunction = jest.fn();
    render(<Button onClick={someFunction}/>);
    const btn = screen.getByRole('button');
    fireEvent.click(btn);
    expect(someFunction).toBeCalledTimes(1);
  })
})
