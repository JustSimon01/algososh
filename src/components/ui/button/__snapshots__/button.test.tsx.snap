// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты кнопки Кнопка c индикацией загрузки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тесты кнопки Кнопка c текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    текст кнопки
  </p>
</button>
`;

exports[`Тесты кнопки Кнопка без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Тесты кнопки Кнопка заблокированна 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
